### PR TITLE
move `applyUpdateAccumulators`

### PR DESCRIPTION
**Pull request by jycor**
This PR deletes the `plan.RowUpdateAccumulator` node, and directly injects an `accumulatorIter` into the rowexec tree in `finalizeIters`.

Doltgres fix: https://github.com/dolthub/doltgresql/pull/909

### CODE REVIEW

File: engine.go [hunk 1]
```diff
@@ -830,7 +852,14 @@ func (e *Engine) executeEvent(ctx *sql.Context, dbName, createEventStatement, us
 		return err
 	}
 
-	iter = finalizeIters(ctx, definitionNode, nil, iter)
+	iter, _, err = finalizeIters(ctx, definitionNode, nil, iter)
+	if err != nil {
+		clearAutocommitErr := clearAutocommitTransaction(ctx)
```
Comment (max-hoffman) [inline engine.go:857]: should this error block be default behavior inside finalizeIters? also should the rest of the finalizers run or not if we hit the error?

**Review by max-hoffman — APPROVED** [review]
First pass it seems OK, I'd need to think longer on the schema thing to know if there's a more optimal approach. If the blast radius is just triggers that seems fine, though.

File: sql/rows.go [hunk 2]
```diff
@@ -201,3 +201,11 @@ func (i *sliceRowIter) Close(*Context) error {
 	i.rows = nil
 	return nil
 }
+
+// CustomRowIter is an extension of RowIter for integrators that wrap RowIters.
+// It allows for analysis rules to inspect the underlying RowIters.
+type CustomRowIter interface {
+	RowIter
+	GetChildIter() RowIter
+	SetChildIter(childIter RowIter) RowIter
```
Comment (max-hoffman) [inline sql/rows.go:210]: Set usually in-place updates, `WithChildIter` would be typical for current behavior

File: sql/rows.go [hunk 3]
```diff
@@ -201,3 +201,11 @@ func (i *sliceRowIter) Close(*Context) error {
 	i.rows = nil
 	return nil
 }
+
+// CustomRowIter is an extension of RowIter for integrators that wrap RowIters.
+// It allows for analysis rules to inspect the underlying RowIters.
+type CustomRowIter interface {
```
Comment (max-hoffman) [inline sql/rows.go:207]: `MutableRowIter`? `UnaryRowIter`? `ChildRowIter`?

**Review by max-hoffman — APPROVED** [review]
LGTM, naming comments

File: sql/rowexec/dml_iters.go [hunk 4]
```diff
@@ -514,6 +516,80 @@ type accumulatorIter struct {
...
+	switch i := iter.(type) {
+	case sql.CustomRowIter:
+		childIter := i.GetChildIter()
+		childIter, sch := AddAccumulatorIter(ctx, childIter)
+		return i.SetChildIter(childIter), sch
+	case *callIter:
+		childIter, sch := AddAccumulatorIter(ctx, i.innerIter)
+		i.innerIter = childIter
+		return i, sch
+	case *beginEndIter:
```
Comment (max-hoffman) [inline sql/rowexec/dml_iters.go:576]: should these two implement the new interface?

File: sql/rows.go (hunk 3 above)
Comment (jycor) [reply sql/rows.go:207]: I like `MutableRowIter`; will change

File: sql/rowexec/dml_iters.go (hunk 4 above)
Comment (jycor) [reply sql/rowexec/dml_iters.go:576]: That would make sense; will change

**Pull request merged**